atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/batch/BatchingTaskRunner.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.keyvalue.dbkvs.impl.batch;
+
+import java.util.function.Function;
+
+public interface BatchingTaskRunner {
+
+    interface BatchingStrategy<InT> {
+        Iterable<? extends InT> partitionIntoBatches(InT collection, int batchSizeHint);
```

Comment:
Is there time when this couldn't return collection? I'd bias toward handling 2B partitions or less and know the partition count.